internal/gatewayapi/validate.go

```diff
@@ -722,3 +725,61 @@ func (t *Translator) validateHostname(hostname string) error {
 
 	return nil
 }
+
+// validateSecretRef checks three things:
+//  1. Dose the secret reference has valid Group and kind
+//  2. If the secret reference is a cross namespace reference, is it permitted by
+//     any ReferenceGrant
+//  3. Does the secret exist
+func (t *Translator) validateSecretRef(
```

Comment:
shouldnt we have a similar function in the code for validating the Gateway listener TLS secret ?

Reply:
Plan to reuse this method for Gateway listener TLS secret as well later in a follow-up PR: 

internal/gatewayapi/validate.go line 288

// TODO zhaohuabing: reuse validateSecretRef